osc-ui/src/main/java/org/osc/core/broker/view/vc/BaseVCWindow.java

```diff
@@ -303,44 +303,40 @@ protected void handleException(final Exception originalException) {
             exception = originalException.getCause();
 
             // TODO this exception leaks large amounts of implementation detail out of the API
-			if (errorType == ErrorType.PROVIDER_EXCEPTION) {
-				if (RestClientException.isConnectException(exception) && isOpenstack()) {
+			if (isOpenstack()) {
+				if (errorType == ErrorType.PROVIDER_EXCEPTION) {
+					if (RestClientException.isConnectException(exception)) {
+						// Keystone Connect Exception
+						contentText = VmidcMessages.getString(VmidcMessages_.VC_CONFIRM_IP, KEYSTONE_CAPTION);
+					}
+				} else if (errorType == ErrorType.PROVIDER_CONNECT_EXCEPTION) {
 					// Keystone Connect Exception
+					// Handle ConnectionException(Not a standard exception) thrown by OpenStack4j
 					contentText = VmidcMessages.getString(VmidcMessages_.VC_CONFIRM_IP, KEYSTONE_CAPTION);
+				} else if (errorType == ErrorType.PROVIDER_AUTH_EXCEPTION) {
+					// Keystone Authentication Exception
+					contentText = VmidcMessages.getString(VmidcMessages_.VC_CONFIRM_CREDS, KEYSTONE_CAPTION);
+				} else if (errorType == ErrorType.CONTROLLER_EXCEPTION) {
+					if (RestClientException.isCredentialError(exception)) {
```

Comment:
move this to a separate private method in order to keep this one shorter and clearer, maybe something like `handleControllerException` and call it inside this case.